components/style/values/generics/length.rs

```diff
@@ -68,13 +68,13 @@ impl<LengthPercentage> LengthPercentageOrAuto<LengthPercentage> {
     }
 }
 
-impl<LengthPercentage> LengthPercentageOrAuto<LengthPercentage>
+impl<U> LengthPercentageOrAuto<U>
```

Comment:
Not much familiar with these conventions, but I think it may be a bit strange to use `U` before `T`?
Maybe use `T` here and rename the existing one to `U`?